addons/mail/static/src/core/web/activity.xml

```diff
@@ -31,9 +31,9 @@
                 <table class="o-mail-Activity-details table table-sm mt-2">
                     <tbody>
                         <tr><td class="text-end fw-bolder">Activity type</td><td><t t-esc="props.activity.activity_type_id[1]"/></td></tr>
-                        <tr><td class="text-end fw-bolder">Created</td><td><t t-esc="displayCreateDate"/> by <t t-esc="props.activity.create_uid[1]"/></td></tr>
+                        <tr><td class="text-end fw-bolder">Created</td><td><t t-esc="props.activity.dateCreateFormatted"/> by <t t-esc="props.activity.create_uid[1]"/></td></tr>
                         <tr><td class="text-end fw-bolder">Assigned to</td><td><t t-esc="props.activity.persona.nameOrDisplayName"/></td></tr>
-                        <tr><td class="text-end fw-bolder">Due on</td><td><t t-esc="props.activity.date_deadline"/></td></tr>
+                        <tr><td class="text-end fw-bolder">Due on</td><td><t t-esc="props.activity.dateDeadlineFormatted"/></td></tr>
```

Comment:
Note: this date was not formatted at all, always display YYYY-MM-DD for every one.